### PR TITLE
e2e_node: remove unnecessary dynamic config changes 

### PR DESCRIPTION
**Pull request by endocrimes**
#### What type of PR is this?

/kind cleanup
/sig node

#### What this PR does / why we need it:

Removes unnecessary uses of dynamickubeletconfig in node_e2e tests, and simplifies ones that do no-op feature enablement.

(n.b I haven't run these yet as I'm working from a Mac today, opening this PR to do so)

#### Which issue(s) this PR fixes:

Related to #105047 

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

### CODE REVIEW

**Review by ffromani — COMMENTED** [review]
thanks @endocrimes ,changes looks good!
once you're happy iwith the PR please ping me, I'll have another pass and add my lgtm

File: test/e2e_node/podresources_test.go
Comment (ehashman) [inline test/e2e_node/podresources_test.go:667]: :100:

**Review by ehashman — COMMENTED** [review]
/priority important-soon

leaving to @fromanirh to LGTM, ping me when you need approval
/assign @fromanirh @ehashman

**Review by ffromani — COMMENTED** [review]
/lgtm
/hold
@matthyx please remove the hold when you are happy

**Comment by matthyx** [comment]
> /lgtm /hold @matthyx please remove the hold when you are happy

Thanks, I'm checking why we have errors in node-kubelet-serial, especially those related to `GracefulNodeShutdown` assumed to be true.

**Comment by fromanirh** [comment]
> > /lgtm /hold @matthyx please remove the hold when you are happy
> 
> Thanks, I'm checking why we have errors in node-kubelet-serial, especially those related to `GracefulNodeShutdown` assumed to be true.

good point, I thought they were unrelated to this PR, but it's worth having a look. I'll also try to find time to look into that later.

**Review by SergeyKanzhelev — COMMENTED** [review]
Please create corresponding test-infra PR

cc: @bobbypage

**Comment by fromanirh** [comment]
> > good point, I thought they were unrelated to this PR, but it's worth having a look. I'll also try to find time to look into that later.
> 
> I did some tests (thanks @endocrimes for your help!) and failures are unrelated to this change. /lgtm /unhold
> 
> (but at some point they need to be fixed) https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=should%20be%20able%20to%20handle%20a%20cancelled%20shutdown

I completely agree. Let's see if all the pending PRs we have in this area helps before to start with a new round of investigation/fixes.

**Review by SergeyKanzhelev — APPROVED** [review]

**Pull request merged**
